src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,99 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks that there is no whitespace before a colon token of a switch case or default.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NoWhitespaceBeforeCaseDefaultColon&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * {
+ *  switch(1) {
+ *     case 1 : // expected violation
+ *       break;
+ *     case 2:
+ *       break;
+ *     default : // expected violation
+ *       break;
+ *  }
+ * }
```

Comment:
example of OK code is needed

Reply:
Done.

Reply:
Done.

Reply:
I dont see it :)

Reply:
I thought the valid switch case (`case 2:`) was enough.
But I guess you want me to add another switch with all cases and default valid.
I'm doing that.

Reply:
It is better to have `// OK` comment on ok code and yes, examples of both tokens. You cannot do this in one switch, so it is better to have 2 of them.

Reply:
Done.